tests/test_range.py

```diff
@@ -724,9 +724,53 @@ def test_until(self):
         self.assert_range('until 20 hours from now', now, datetime(2017, 6, 17, 15))
         self.assert_range('until 45 minutes from now', now, datetime(2017, 6, 16, 20, 22))
         self.assert_range('until 45 seconds from now', now, datetime(2017, 6, 16, 19, 38, 7))
+
+        # TODO "Until yesterday" etc: error or guess or infinity or unknown
+        # TODO "Until the start of today" vs "Until the end of today"
+
+    def test_by(self):
+        now = datetime.now()
+        self.assert_range('by next may', now, datetime(2018, 6, 1))
+        self.assert_range('by next Friday', now, datetime(2017, 6, 24))
+        self.assert_range('by next Saturday', now, datetime(2017, 6, 18))
+        self.assert_range('by next Thursday', now, datetime(2017, 6, 23))
+        self.assert_range('by next year', now, datetime(2019, 1, 1))
+        self.assert_range('by next month', now, datetime(2017, 8, 1))
+        self.assert_range('by next week', now, datetime(2017, 6, 26))
```

Comment:
Thus, based on our convo earlier, this should default to datetime(2017, 6, 24, 0,0,0).